--- darktable-2.2.5/src/common/dynload.c~	2017-05-28 14:22:21.000000000 +0000
+++ darktable-2.2.5/src/common/dynload.c	2017-08-04 12:26:09.762458898 +0000
@@ -44,6 +44,7 @@
   dt_gmodule_t *module = NULL;
   GModule *gmodule;
   char *name;
+  char name2[8192];
 
   if(strchr(library, '/') == NULL)
   {
@@ -54,7 +55,12 @@
     name = g_strdup(library);
   }
 
-  gmodule = g_module_open(name, G_MODULE_BIND_LAZY | G_MODULE_BIND_LOCAL);
+  gmodule = NULL;
+  sprintf(name2, "%s.avx2", name);
+  if (__builtin_cpu_supports("avx2"))
+  	gmodule = g_module_open(name2, G_MODULE_BIND_LAZY | G_MODULE_BIND_LOCAL);
+  if (!gmodule)
+	  gmodule = g_module_open(name, G_MODULE_BIND_LAZY | G_MODULE_BIND_LOCAL);
 
   if(gmodule != NULL)
   {
