diff --git a/src/common/dynload.c b/src/common/dynload.c
index 3e4834c..412286a 100644
--- a/src/common/dynload.c
+++ b/src/common/dynload.c
@@ -46,6 +46,7 @@ dt_gmodule_t *dt_gmodule_open(const char *library)
   dt_gmodule_t *module = NULL;
   GModule *gmodule;
   char *name;
+  char name2[8192];
 
   if(strchr(library, '/') == NULL)
   {
@@ -56,7 +57,12 @@ dt_gmodule_t *dt_gmodule_open(const char *library)
     name = g_strdup(library);
   }
 
-  gmodule = g_module_open(name, G_MODULE_BIND_LAZY | G_MODULE_BIND_LOCAL);
+  gmodule = NULL;
+  sprintf(name2, "%s.avx2", name);
+  if(__builtin_cpu_supports("avx2"))
+    gmodule = g_module_open(name2, G_MODULE_BIND_LAZY | G_MODULE_BIND_LOCAL);
+  if(!gmodule)
+    gmodule = g_module_open(name, G_MODULE_BIND_LAZY | G_MODULE_BIND_LOCAL);
 
   if(gmodule != NULL)
   {
